modules/29_CES_parameters/calibrate/preloop.gms

```diff
@@ -257,6 +257,35 @@ if (sm_tmp eq 1,
   abort "some prices are negative. See log file";
 );
 
+*** Check for labour xi < 0.1 and scale all ppf prices to prevent it.
+p29_ppf_price_multiplicator(t,regi_dyn29(regi))
+  = pm_cesdata(t,regi,"inco","quantity")
+  / sum(ppf$( NOT sameas(ppf,"lab") ),
+      pm_cesdata(t,regi,ppf,"quantity")
+    * pm_cesdata(t,regi,ppf,"price")
+    );
+
+if (smax((t,regi_dyn29(regi)), 1 / p29_ppf_price_multiplicator(t,regi)) gt 0.9,
+  put logfile, ">>> Scaling ppf prices to ensure xi labour >= 0.1 <<<" /;
+  loop ((t,regi_dyn29(regi))$( 1 / p29_ppf_price_multiplicator(t,regi) gt 0.9 ),
+    put p29_ppf_price_multiplicator.tn(t,regi), " = ";
+    put p29_ppf_price_multiplicator(t,regi) /;
+    
+    pm_cesdata(t,regi,ppf,"price")$( NOT sameas(ppf,"lab") )
+    = pm_cesdata(t,regi,ppf,"price")
+    * p29_ppf_price_multiplicator(t,regi);
+  );
+
+$ontext
+  if (%c_CES_calibration_iteration% gt 3,   !! c_CES_calibration_iteration
+    execute_unload "abort.gdx";
+    abort "Too low labour xi after third CES iteration.  CALIBRATION PANIC!";
+  );
+$offtext
+
+  display "after ppf price multiplicator",
+    p29_ppf_price_multiplicator, pm_cesdata;
+);
```

Comment:
is there a mechanism to stop this price_multiplicator? How to ensure that other consistency checks do not fail because of the price shift?

Reply:
> is there a mechanism to stop this price_multiplicator?

See
> Theoretically, this should not occur in later calibration iterations.
> But for now, the test for that is deactivated to observe the
> calibration behaviour.

and [lines 279-84](
https://github.com/remindmodel/remind/pull/763/files/ce3ae4b5caafe7b99087a66069626ef3697ab60c#diff-54a5960804a16adc77b7ddc96e32be89e442abbd1f89b0fa5cd8e7c5a1b174b7R279-R284)
```
$ontext
  if (%c_CES_calibration_iteration% gt 3,   !! c_CES_calibration_iteration
    execute_unload "abort.gdx";
    abort "Too low labour xi after third CES iteration.  CALIBRATION PANIC!";
  );
$offtext
```

Reply:
> How to ensure that other consistency checks do not fail because of the price shift?

Prices are modified directly after derivation, so all calculations the consistency checks are checking are based on these modified prices.

Reply:
ah, you plan to activate lines 279-84, thanks